lld/Common/ErrorHandler.cpp

```diff
@@ -335,3 +335,21 @@ void ErrorHandler::fatal(const Twine &msg) {
   error(msg);
   exitLld(1);
 }
+
+SyncStream::~SyncStream() {
+  os.flush();
+  switch (level) {
+  case DiagLevel::Log:
+    e.log(buf);
```

Comment:
I wonder if it'd be helpful to have the log-msg logged instantly to the handler rather than waiting till the end?
(The messages might get dropped if lld crashed before the dtor gets to run?)

Reply:
If we make `Warn(ctx) << "a" << "b"` print "a" immediately, "a" & "b" diagnostics from the threads might interleave. SyncStream is to ensure that only complete diagnostics can interleave, not the fragments.